examples/nfct_test.rs

```diff
@@ -5,166 +5,194 @@ extern crate lang_items;
 
-    writeln!(console, "****************************************").unwrap();
-    writeln!(console, "nfct_test application is installed").unwrap();
-    writeln!(
-        console,
-        "Clock frequency: {} Hz",
-        timer.clock_frequency().hz()
-    )
-    .unwrap();
+    /// Helper function to write on console the received packet.
+    fn print_rx_buffer(buf: &mut [u8]) {
+        let mut console = Console::new();
+        write!(console, "RX:").unwrap();
+        if let Some((last, bytes)) = buf.split_last() {
+            for byte in bytes {
+                write!(console, " {:02x?}", byte).unwrap();
+            }
+            writeln!(console, " {:02x?}", last).unwrap();
+        }
+        console.flush();
+    }
 
-    let mut state = State::Disabled;
-    // Variable to count the change in the tag's state
-    let mut state_change_cntr = 0;
-    loop {
-        match state {
-            State::Enabled => {
-                let mut rx_buf = [0; 256];
-                loop {
-                    match NfcTag::receive(&mut rx_buf) {
-                        Ok(RecvOp {
-                            recv_amount: amount,
-                            ..
-                        }) => print_rx_buffer(&mut rx_buf, amount),
-                        Err(TockError::Command(CommandError {
-                            return_code: -4, /* EOFF: Not Ready */
-                            ..
-                        })) => (),
-                        Err(TockError::Command(CommandError {
-                            return_code: value, ..
-                        })) => writeln!(console, " -- Err({})!", value).unwrap(),
-                        Err(_) => writeln!(console, " -- RX Err").unwrap(),
+    /// Function to identify the time elapsed for a transmission request.
+    fn bench_transmit(console: &mut Console, timer: &Timer, title: &str, mut buf: &mut [u8]) {
+        let amount = buf.len();
+        let start = Timestamp::<f64>::from_clock_value(timer.get_current_clock().flex_unwrap());
+        match NfcTag::transmit(&mut buf, amount) {
+            Ok(_) => (),
+            Err(_) => writeln!(Console::new(), " -- tx error!").unwrap(),
+        }
+        let end = Timestamp::<f64>::from_clock_value(timer.get_current_clock().flex_unwrap());
+        let elapsed = (end - start).ms();
+        writeln!(
+            console,
+            "{}\n{:.2} ms elapsed for {} bytes ({:.2} kbit/s)",
+            title,
+            elapsed,
+            amount,
+            (amount as f64) / elapsed * 8.
+        )
+        .unwrap();
+        console.flush();
+    }
+
+    fn receive_packet(console: &mut Console, mut buf: &mut [u8; 256]) {
+        match NfcTag::receive(&mut buf) {
+            Ok(RecvOp {
+                recv_amount: amount,
+                ..
+            }) => {
+                if amount > 0 && amount <= buf.len() {
+                    print_rx_buffer(&mut buf[..amount]);
+                }
```

Comment:
FYI: If you want to handle the cases with different amounts too, you could move the `if` before `=>` (maybe even in 2 separate arms). Depends on if you want to silently ignore the 0 check.

If you do, I'd probably remove the check here, pass an empty slice into the function, and move the `console` declaration in `print_rx_buffer` inside the `if let`. That way, you have a more general function (because it can also handle empty slices). And a little bit easier code here.

Reply:
I don't understand where `if let` should go as I'm matching on some errors after the `Ok()`. I'm ignoring the 0 as I won't be printing anything inside of the function, so I don't want to make an unnecessary function call.

Reply:
Ignoring is perfectly fine. What I'm saying is: Remove the `amount > 0` check here, and then change `print_rx_buffer` to gracefully handle the empty slice:

        if let Some((last, bytes)) = buf.split_last() {
          let mut console = Console::new();
          write!(console, "RX:").unwrap();
            for byte in bytes {
                write!(console, " {:02x?}", byte).unwrap();
            }
            writeln!(console, " {:02x?}", last).unwrap();
            console.flush();
        }